odoo/addons/base/res/res_partner.py

```diff
@@ -625,7 +625,7 @@ def name_create(self, name):
             a name, the name will have the email value.
             If 'force_email' key in context: must find the email address. """
         default_type = self._context.get('default_type')
-        if default_type and default_type not in self._fields['type'].selection:
+        if default_type and default_type not in self._fields['type'].get_values(self):
```

Comment:
@nboulif get_values() expect to receive an `Environment` as parameter, see https://github.com/odoo/odoo/blob/11.0/odoo/fields.py#L1813-L1821
```suggestion
        if default_type and default_type not in self._fields['type'].get_values(self.env):
```